substrafl/remote/register/register.py

```diff
@@ -320,3 +323,118 @@ def register_algo(
             )
         )
         return key
+
+
+def _check_metric_function(metric_function: typing.Callable):
+    """Function to check the type and the signature of a given score function.
+
+    Args:
+        metric_function (typing.Callable): function to check.
+
+    Raises:
+        exceptions.MetricFunctionTypeError: metric_function must be of type "function"
+        exceptions.MetricFunctionSignatureError: metric_function must ONLY contains
+            datasamples and predictions_path as parameters
+    """
+
+    if not inspect.isfunction(metric_function):
+        raise exceptions.MetricFunctionTypeError("The metric_function() must be of type function.")
+
+    signature = inspect.signature(metric_function)
+    parameters = signature.parameters
+
+    if "datasamples" not in parameters:
+        raise exceptions.MetricFunctionSignatureError(
+            "The metric_function() function must contain datasamples as parameter."
+        )
+    elif "predictions_path" not in parameters:
+        raise exceptions.MetricFunctionSignatureError(
+            "The metric_function() function must contain predictions_path as parameter."
+        )
+    elif len(parameters) != 2:
+        raise exceptions.MetricFunctionSignatureError(
+            """The metric_function() function must ONLY contains datasamples and predictions_path as
+            parameters."""
+        )
+
+
+def add_metric(
+    client: substra.Client,
+    permissions: substra.sdk.schemas.Permissions,
+    dependencies: Dependency,
+    metric_function: typing.Callable,
+    metric_name: typing.Optional[str] = None,
+) -> str:
+    """This function add a metric to the Substra platform using the given score function as the
```

Comment:
Nitpick:
```suggestion
    """This function adds a metric to the Substra platform using the given score function as the
```

Or perhaps just "adds" since we know it's a function, or "add" (imperative) as in Git commit messages. Idk what the SubstraFL style is